Core/Config.cpp

```diff
@@ -1581,6 +1595,24 @@ bool Config::Save(const char *saveReason) {
 		Section *control = iniFile.GetOrCreateSection("Control");
 		control->Delete("DPadRadius");
 
+		// Migrate from wrong section...
```

Comment:
I think this is okay, but I'll note that it's ideal to check some ini value at a default (like check if DoubleTapGesture is -1 and set to 0 the actual default, or something) around the defaulting, and not delete.

The reason is that sometimes we ask users to test against old versions to validate a bug, and it's convenient when their old settings just work and are kept.  The current way would result in this both resetting to defaults in the old versions, and then subsequently resetting to defaults in the new versions too (because they'd exist again.)

I know DPadRadius is doing this, though...

-[Unknown]

Reply:
I don't really get what you mean with the default check.

If we want to allow old versions there are 2 options I can think:
- Have both settings in the ini and update both, that's kinda ugly
- Leave the settings where they are and just change the profile save/load logic to get them as well

If we want to allow only old version settings if the user used an old version already:
- Update both if the keys are already present

Not sure what's best or if there are other ways.

Reply:
Sorry for the delay, I've had a complex week and missed this notification.

I meant something like:

```c++
ConfigSetting("DoubleTapGesture", &g_Config.iDoubleTapGesture, -1, true, true),

...

if (g_Config.iDoubleTapGesture == -1) {
	// The -1 value is invalid, we use it entirely to know that the control settings are defaults.
	g_Config.iDoubleTapGesture = 0;

	loadOldControlSettings();
}
```

That way, if `DoubleTapGesture = 0` (or any other value) already exists, it won't port the old settings over.  If it doesn't exist yet, it will copy them over (and very briefly use -1 during the loading process, but other code should only see 0 and other valid values.)

Checking this way would deterministically tell us if it's time to port the settings, but it would also allow us to keep the old settings without deleting them.  No need to update the old ones.

Alternatively, we could do effectively the same with a "SettingsVersion" key under `[General]`, which might be easier to understand for people reading the code and future changes.

-[Unknown]

Reply:
Oh, now I get it.

Went for the "SettingsVersion" way because we have bool to port as well and can't set them to invalid value :)